### PR TITLE
Disable enforcing JSDoc internals for functions that do not have JSDocs

### PR DESCRIPTION
**Pull request by zachintosh**
From this point on, only functions that _have_ JSDocs will have the remainder of the rules applied.
Update eslint-plugin-jsdoc: ^20 --> ^29.

Release version 2.1.0.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ module.exports = {
     // 'eslint-plugin-bestpractices',
     // 'eslint-plugin-deprecate',
     'eslint-plugin-html',
-    'eslint-plugin-jsdoc',
+    // 'eslint-plugin-jsdoc',
```
Comment (zachintosh) [inline index.js:58]:
```suggestion
```

**Comment by skye2k2** [comment]
I would recommend updating the README to stick `eslint-plugin-jsdoc` in a removed section with a note as to why, updating the dependency version in the usage section, running the `test:update` script, and committing the corresponding files.

This code does not yet have a build to check things like that, sorry.

File: index.js [hunk 2]
```diff
@@ -67,35 +66,6 @@ module.exports = {
...
-    'jsdoc/newline-after-description': 'off',
-    'jsdoc/no-types': 'off',
-    'jsdoc/no-undefined-types': 'off', // 2020-01-23: This was broken in eslint-plugin-jsdoc#8 in 2019-06, and hasn't gotten much better. Disabled, for now. Check back later.
-
-    'jsdoc/require-description-complete-sentence': 'off',
-    'jsdoc/require-description': 'warn',
-    'jsdoc/require-example': 'off',
-    'jsdoc/require-file-overview': 'off',
-    'jsdoc/require-hyphen-before-param-description': 'warn',
-    'jsdoc/require-jsdoc': 'warn',
```
Comment (skye2k2) [inline index.js:88]: I, personally, feel that setting this rule to `off` would give you the freedom you desired, but I have not tested this theory.

Comment (zachintosh) [reply index.js:88]: I see. So it wouldn't require us to add jsdocs, but as soon as you do, the other jsdoc rules would kick in. I'd be good with that.

Comment (skye2k2) [reply index.js:88]: winwinwin

**Pull request merged**